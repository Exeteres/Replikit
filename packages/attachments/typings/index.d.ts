export * from "./entities/attachment";

export * from "./extensions/storage";
