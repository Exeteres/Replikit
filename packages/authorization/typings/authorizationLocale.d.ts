export interface AuthorizationLocale {
    accessDenied: string;
}
