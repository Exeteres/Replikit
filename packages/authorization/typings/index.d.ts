export * from "./authorizarionLocale";
export * from "./extensions/i18n";
