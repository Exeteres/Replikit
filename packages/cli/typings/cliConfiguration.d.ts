export interface CliConfiguration {
    projectRoot?: string;
    modules?: string[];
}
