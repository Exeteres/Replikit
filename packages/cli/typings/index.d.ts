export * from "./packageConfig";
export * from "./cliConfiguration";
export * from "./extensions/core";
