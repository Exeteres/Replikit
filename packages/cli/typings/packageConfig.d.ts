export interface PackageConfig {
    name: string;
    version: string;
    license: string;
    private?: boolean;
    workspaces?: string[];
    scripts?: Record<string, string>;
    dependencies?: Record<string, string>;
    devDependencies?: Record<string, string>;
}
