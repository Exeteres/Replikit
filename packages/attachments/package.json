{
    "name": "@replikit/attachments",
    "version": "0.1.0",
    "license": "MIT",
    "scripts": {
        "build": "ttsc -p tsconfig.build.json"
    },
    "dependencies": {
        "@replikit/storage": "^0.1.0"
    }
}
