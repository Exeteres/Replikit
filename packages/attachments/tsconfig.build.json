{
    "extends": "../../tsconfig.build.json",
    "compilerOptions": {
        "outDir": "dist",
        "paths": {
            "@replikit/attachments": ["attachments/src"]
        }
    },
    "include": ["src"]
}
