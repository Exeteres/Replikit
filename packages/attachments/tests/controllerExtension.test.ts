import "@replikit/attachments";
import { ControllerExtension } from "@replikit/attachments";
import {
    TestController,
    DatabaseTestManager,
    TestControllerOptions
} from "@replikit/test-utils";
import { AttachmentType } from "@replikit/core";
import { SendedMessage } from "@replikit/core/typings";
import { Attachment } from "@replikit/attachments/typings";

let testManager: DatabaseTestManager;

beforeEach(() => {
    testManager = new DatabaseTestManager();
    return testManager.connect();
});

afterEach(() => {
    return testManager.close();
});

function createExtension(options?: TestControllerOptions): ControllerExtension {
    const extension = (new TestController(
        options
    ) as unknown) as ControllerExtension;
    extension.connection = testManager.connection;
    return extension;
}

function createSendedMessage(): SendedMessage {
    return {
        id: 1,
        attachments: [
            {
                id: "1",
                uploadId: "uploadId",
                origin: {
                    id: "123",
                    source: "123",
                    controllerName: "another",
                    type: AttachmentType.Photo
                }
            }
        ],
        metadata: { firstAttachment: true, messageIds: [1] }
    };
}

describe("ControllerExtension", () => {
    it("should do nothing if controller don't return sended attachment", async () => {
        const controller = createExtension();
        await controller.sendMessage(1, {
            attachments: [{ id: "123", type: AttachmentType.Photo }],
            forwarded: [],
            tokens: []
        });

        const collection = testManager.connection //
            .getRawCollection<Attachment>("attachments");

        const attachments = await collection.find().toArray();
        expect(attachments).toHaveLength(0);
        expect.assertions(1);
    });

    it("should do nothing if controllerName matches the controller", async () => {
        const message = createSendedMessage();
        const controller = createExtension({ sendedMessage: message });
        await controller.sendMessage(1, {
            attachments: [
                {
                    id: "123",
                    type: AttachmentType.Photo,
                    controllerName: "test"
                }
            ],
            tokens: [],
            forwarded: []
        });

        const collection = testManager.connection //
            .getRawCollection<Attachment>("attachments");

        const attachments = await collection.find().toArray();
        expect(attachments).toHaveLength(0);
        expect.assertions(1);
    });

    it("should save attachment returned from controller", async () => {
        const message = createSendedMessage();
        const controller = createExtension({ sendedMessage: message });
        await controller.sendMessage(1, {
            attachments: [
                {
                    id: "123",
                    controllerName: "another",
                    type: AttachmentType.Photo
                }
            ],
            tokens: [],
            forwarded: []
        });

        const collection = testManager.connection //
            .getRawCollection<Attachment>("attachments");

        const attachments = await collection.find().toArray();
        expect(attachments).toHaveLength(1);
        expect(attachments[0]).toMatchSnapshot();
        expect.assertions(2);
    });

    it("should append the source to existing attachment", async () => {
        const collection = testManager.connection //
            .getRawCollection<Attachment>("attachments");

        await collection.insertOne({
            _id: {
                controller: "another",
                localId: "123"
            },
            sources: [
                {
                    controller: "anotherTest",
                    localId: "1",
                    uploadId: "uploadId"
                }
            ]
        });

        const message = createSendedMessage();
        const controller = createExtension({ sendedMessage: message });
        await controller.sendMessage(1, {
            attachments: [
                {
                    id: "123",
                    controllerName: "another",
                    type: AttachmentType.Photo
                }
            ],
            tokens: [],
            forwarded: []
        });

        const attachments = await collection.find().toArray();
        expect(attachments).toHaveLength(1);
        expect(attachments[0]).toMatchSnapshot();
        expect.assertions(2);
    });

    it("should use the stored attachment source", async () => {
        const collection = testManager.connection //
            .getRawCollection<Attachment>("attachments");

        await collection.insertOne({
            _id: {
                controller: "another",
                localId: "123"
            },
            sources: [
                {
                    controller: "test",
                    localId: "1",
                    uploadId: "uploadId"
                }
            ]
        });

        const controller = createExtension();
        const source = await controller["resolveSource"](1, {
            id: "123",
            controllerName: "another",
            type: AttachmentType.Photo
        });
        expect(source).toBe("uploadId");
    });
});
