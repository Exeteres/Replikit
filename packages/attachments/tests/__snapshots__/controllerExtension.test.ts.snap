// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControllerExtension should append the source to existing attachment 1`] = `
Object {
  "_id": Object {
    "controller": "another",
    "localId": "123",
  },
  "sources": Array [
    Object {
      "controller": "anotherTest",
      "localId": "1",
      "uploadId": "uploadId",
    },
    Object {
      "controller": "test",
      "localId": "1",
      "uploadId": "uploadId",
    },
  ],
}
`;

exports[`ControllerExtension should save attachment returned from controller 1`] = `
Object {
  "_id": Object {
    "controller": "another",
    "localId": "123",
  },
  "sources": Array [
    Object {
      "controller": "test",
      "localId": "1",
      "uploadId": "uploadId",
    },
  ],
}
`;
